client/layers.go

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2021 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+)
+
+type MergeLayerOptions struct {
+	Layer string
```

Comment:
This is related to the question you've asked during our call: Go is statically typed, so we need to think well which parameters we have here to make it future proof. We've done the "format" trick in the payload to enable us to send it as structured data, so we need to ponder for a moment that might be here eventually. I suspect "Layer" would be the most natural name for the structured type, which means we need to move this out onto a separate field. Maybe LayerData as a []byte, and document it as being yaml?  Then, in the future we may introduce "Format", if necessary, or just "Layer" as something strongly typed.